### PR TITLE
Fixes for a few finegrained token issues

### PR DESCRIPTION
**Pull request by dustin-decker**
### Description:
Fixes a few issues encountered with finegrained tokens.

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/analyzer/analyzers/github/finegrained/finegrained.go [hunk 1]
```diff
@@ -1336,31 +1331,32 @@ func PrintFineGrainedToken(cfg *config.Config, info *common.SecretInfo) {
...
+		keys = append(keys, val)
 	}
 	// Sort the slice
 	sort.Strings(keys)
 
-	for _, key := range keys {
-		value := accessMap[key]
-		if value == Invalid {
+	for idx, permStr := range keys {
+		perm := accessMap[idx]
```
Comment (mcastorina) [inline pkg/analyzer/analyzers/github/finegrained/finegrained.go:1350]: If we're relying on the index on line 1350, I think we also need to remove the `sort` on 1347.

File: pkg/analyzer/analyzers/github/finegrained/finegrained.go [hunk 2]
```diff
@@ -428,11 +428,6 @@ func getDependabotAlertsPermission(client *gh.Client, repo *gh.Repository, curre
 	// Risk: Extremely Low
 	// GET /repos/{owner}/{repo}/dependabot/alerts
 	_, resp, err := client.Dependabot.ListRepoAlerts(context.Background(), *repo.Owner.Login, *repo.Name, &gh.ListAlertsOptions{})
-	if err != nil {
-		if !strings.Contains(err.Error(), "disabled") {
-			return NoAccess, err
-		}
-	}
```
Comment (mcastorina) [inline pkg/analyzer/analyzers/github/finegrained/finegrained.go:435]: It looks like this `err` is completely unchecked now, which seems wrong.

File: pkg/analyzer/analyzers/github/finegrained/finegrained.go (hunk 1 above)
Comment (dustin-decker) [reply pkg/analyzer/analyzers/github/finegrained/finegrained.go]: Fixed, just accessing directly now

File: pkg/analyzer/analyzers/github/finegrained/finegrained.go (hunk 2 above)
Comment (dustin-decker) [reply pkg/analyzer/analyzers/github/finegrained/finegrained.go:435]: It's returned in the default case

**Review by mcastorina — APPROVED** [review]

**Pull request merged**